src/types/entry.rs

```diff
@@ -18,20 +18,36 @@ use crate::*;
 /// Entry is the file/dir entry returned by `Lister`.
 #[derive(Clone, Debug)]
 pub struct Entry {
+    /// Path of the entry.
     path: String,
-    meta: Metadata,
+
+    /// Optional cached metadata
+    cached_metadata: Option<Metadata>,
```

Comment:
We can use `metadata` directly here

Reply:
@Xuanwo ?

Reply:
The semantice is clean enough, let's use `metadata`.